src/main/java/pages/DefectDetailsPage.java

```diff
@@ -89,6 +94,41 @@ public void selectVerticalValue(String id) {
         findElementById(id).click();
     }
 
+    public void setReferenceNumber(String referenceNumber) {
+
```

Comment:
For all these fields you can use xpath locator wehre you can use as reference elements that have certain attributes that uniquely identifiy them in the page. For example the text field after "REFERENCE NUMBER" can be accessed using this xpath '//XCUIElementTypeStaticText[@value="REFERENCE NUMBER"]/../following-sibling::XCUIElementTypeOther/XCUIElementTypeTextField'. @jgimber-deloitte Is is up to you to decide the approach you prefer.

Reply:
OK thanks for the tip - will include it in the next batch of changes.